### PR TITLE
nixos: unify virtual console options

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change

As per commit message:

This commit moves all the virtual console related options
to a dedicated config/console.nix NixOS module.

Currently most of these are defined in config/i18n.nix
with a "console" prefix like `i18n.consoleFont`,
`i18n.consoleColors` or under `boot` and are implemented
in tasks/kbd.nix.
Since they have little to do with actual internationalisation
and are (informally) in an attrset already, it makes sense to
move them to a specific module.

Also, this commit changes the console colors implementation
to use the kernel parameters instead of relying on terminal
escape sequences. This means the palette is applied by the
kernel itself with no for custom code running in the initrd
and works for virtual terminals (not only tty0).

###### Things done

- [ ] Tested via one or more NixOS tests (not aware of any one specific)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @vcunat @abbradar @greedy @the-kenny

### CODE REVIEW

**Comment by pstch** [comment]
- [ ] changes are backward compatible : _renamed options_
- [X] renamed options are declared with `mkRenamedOptionModule`
- [X] changes that are not backward compatible are documented in release notes
- [ ] module tests succeed on linux amd64 _not applicable_
- [X] options types are appropriate : _no change_
- [X] options description is set : _no change_
- [X] options example is provided : _no change_
- [X] documentation affected by the changes is updated : _manual not affected_

Suggestions:
 - as this PR changes important options that are defined in many configurations, it may be better to isolate it from the changes to the color palette application

**Comment by rnhmjoj** [comment]
Thank you for the review, @pstch.

> as this PR changes important options that are defined in many configurations, it may be better to isolate it from the changes to the color palette application

I thought about it but I made these changes simultaneously so it would require some nontrivial history rewriting.

**Comment by pstch** [comment]
I missed something in my review:
 - The old definition of `i18n.consoleKeyMap` has not been removed (in `i18n.nix`)
 - The XML of the release note is broken.

I'm rebasing your commits, fixing both of these issues, and will push my branch so that you can use it if you want. I also removed the changes to the color palette application by diffing the two files, so that we can add them in another PR.

**Comment by pstch** [comment]
@rnhmjoj I [rebased your PR](https://github.com/pstch/nixpkgs/tree/feature/unified-console-options) on recent master, applying the following steps:
 - fixing the XML in the release notes, removing the commit for indentation that doesn't apply anymore
 - removing the changes to the color palette application
 - removing the old definition of `i18n.consoleKeyMap` in `i18n.nix`

I have tested these commits on my machine by rebasing them on 19.09.

**Comment by rnhmjoj** [comment]
@pstch Thank you. I fixed the merge conflict and followed your suggestions.

**Comment by rnhmjoj** [comment]
Fixed yet another conflict.

**Comment by rnhmjoj** [comment]
This time it was me. I wonder why git can't automaticallt merge this...

File: nixos/modules/config/console.nix [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        ];
+      })
+
+      (mkIf cfg.earlySetup {
+        boot.initrd.extraUtilsCommands = ''
+          mkdir -p $out/share/consolefonts
+          ${if substring 0 1 cfg.font == "/" then ''
+            font="${cfg.font}"
+          '' else ''
+            font="$(echo ${consoleEnv}/share/consolefonts/${cfg.Font}.*)"
```
Comment (Infinisil) [inline nixos/modules/config/console.nix]: Should be `cfg.font`

**Review by infinisil — CHANGES_REQUESTED** [review]

File: nixos/modules/config/console.nix (hunk 1 above)
Comment (rnhmjoj) [reply nixos/modules/config/console.nix]: Thank you, I missed it.

**Comment by rnhmjoj** [comment]
@Infinisil What to do you think about adding a new top level attribute? I'm not too keen but I didn't find a suitable existing one.

**Review by infinisil — APPROVED** [review]
Sounds like a good idea to me, it really doesn't fit anywhere else.

I tested this in a VM, seems to work fine, let's merge :)

**Comment by rnhmjoj** [comment]
Thank you!

**Pull request merged**